cfme/infrastructure/host.py

```diff
@@ -403,8 +364,154 @@ def _select_rows(indexes):
+               host_platform=None, custom_ident=None, ipmi_address=None, mac_address=None,
+               ipmi_credentials=None, cancel=False, validate_credentials=False):
+        """Creates a host in the UI.
+
+        Args:
+           cancel (bool): Whether to cancel out of the creation. The cancel is done after all the
+               information present in the Host has been filled in the UI.
+           validate_credentials (bool): Whether to validate credentials - if True and the
+               credentials are invalid, an error will be raised.
+        """
+        view = navigate_to(self, "Add")
+        view.fill({
+            "name": name,
+            "hostname": hostname or ip_address,
+            "host_platform": host_platform,
+            "custom_ident": custom_ident,
+            "ipmi_address": ipmi_address,
+            "mac_address": mac_address
+        })
+        if credentials is not None:
+            view.endpoints.default.fill(credentials.view_value_mapping)
+            if validate_credentials:
+                view.endpoints.default.validate_button.click()
+        if ipmi_credentials is not None:
+            view.endpoints.ipmi.fill(ipmi_credentials.view_value_mapping)
+            if validate_credentials:
+                view.endpoints.ipmi.validate_button.click()
+        if not cancel:
+            view.add_button.click()
+            flash_message = 'Host / Node " {}" was added'.format(name)
+        else:
+            view.cancel_button.click()
+            flash_message = "Add of new Host / Node was cancelled by the user"
+        host = self.instantiate(name=name, hostname=hostname, ip_address=ip_address,
+                                custom_ident=custom_ident, host_platform=host_platform,
+                                ipmi_address=ipmi_address, mac_address=mac_address,
+                                credentials=credentials, ipmi_credentials=ipmi_credentials,
+                                provider=provider)
+        view = host.create_view(HostsView)
+        assert view.is_displayed
+        view.flash.assert_success_message(flash_message)
+        return host
+
+    def all(self, provider):
+        """returning all Object Store Objects"""
+        view = navigate_to(self, 'All')
+        view.toolbar.view_selector.select("Grid View")
+        hosts = [self.instantiate(name=item, provider=provider)
+                 for item in view.entities.entity_names]
+        return hosts
+
+    def run_smartstate_analysis(self, *hosts):
+        hosts = list(hosts)
+        checked_hosts = list()
+        view = navigate_to(self, 'All')
+
+        for host in hosts:
+            try:
+                view.entities.get_entity(by_name=host.name, surf_pages=True).check()
+                checked_hosts.append(host)
+            except ItemNotFound:
+                raise ValueError('Could not find host {} in the UI'.format(host.name))
```

Comment:
**REQUIRED**
```python
hosts = list(hosts)
checked_hosts = list()
view = navigate_to(self, 'All')
for host in hosts:
    try:
        view.entities.get_entity(by_name=host.name, surf_pages=True).check()
        checked_hosts.append(host)
    except ItemNotFound:
        raise ValueError('Could not find host {} in the UI'.format(host.name))
```
Made a separate method for that in order to not repeat the code.

Reply:
done

Reply:
I think `ItemNotFound` should be raised here